File: sdk/arena-java-sdk/src/main/java/com/github/kubeflow/arena/model/evaluate/EvaluateJob.java
package com.github.kubeflow.arena.model.evaluate;

import java.util.List;

public class EvaluateJob {

    private String name;
    private List<String> args;
    private String command;

    public EvaluateJob(String name, List<String> args, String command) {
        this.name = name;
        this.args = args;
        this.command = command;
    }

    public String name() {
        return this.name;
    }

    public List<String> getArgs() {
        return this.args;
    }

    public String getCommand() {
        if (this.command == null) {
            return "";
        }
        return this.command;
    }

}

File: pkg/apis/evaluate/evaluatejob.go
package evaluate

type baseJob struct {
	name string
	args interface{}
}

type EvaluateJob struct {
	baseJob
}

func newBaseJob(name string, args interface{}) baseJob {
	return baseJob{
		name: name,
		args: args,
	}
}

func (b *baseJob) Name() string {
	return b.name
}

func (b *baseJob) Args() interface{} {
	return b.args
}

func NewEvaluateJob(name string, args interface{}) *EvaluateJob {
	return &EvaluateJob{
		baseJob: newBaseJob(name, args),
	}
}
